#VRML_SIM V6.0 utf8
WorldInfo {
  info [
    "Description"
    "Author: first name last name <e-mail>"
    "Date: DD MMM YYYY"
  ]
  basicTimeStep 16
  contactProperties [
    ContactProperties {
      material2 "nofriction"
      coulombFriction 0.001
    }
  ]
}
Viewpoint {
  orientation -0.33742 0.941341 0.00502471 1.23511
  position 0.283276 0.415468 0.15498
}
Background {
  skyColor [
    0.4 0.7 1
  ]
}
DirectionalLight {
  ambientIntensity 0.5
  direction 1 -2 1
  castShadows TRUE
}
DEF FLOOR Solid {
  translation -0.5 0 -0.5
  children [
    Shape {
      geometry ElevationGrid {
        color Color {
          color [
            0.43653 0.598901 0.288273
            1 1 0
          ]
        }
        colorPerVertex FALSE
        xDimension 11
        xSpacing 0.1
        zDimension 11
        zSpacing 0.1
      }
    }
  ]
  boundingObject Transform {
    translation 0.5 0 0.5
    children [
      Plane {
      }
    ]
  }
}
DEF WALL_1 Solid {
  translation 0.5 0.025 0
  children [
    DEF WALL_SHAPE Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0.12549 0.368627 0.729412
        }
      }
      geometry Box {
        size 0.02 0.05 0.98
      }
    }
  ]
  boundingObject USE WALL_SHAPE
}
DEF WALL_2 Solid {
  translation -0.5 0.025 0
  children [
    USE WALL_SHAPE
  ]
  boundingObject USE WALL_SHAPE
}
DEF WALL_3 Solid {
  translation 0 0.025 -0.5
  rotation 0 1 0 1.5708
  children [
    USE WALL_SHAPE
  ]
  boundingObject USE WALL_SHAPE
}
DEF WALL_4 Solid {
  translation 0 0.025 0.5
  rotation 0 1 0 1.5708
  children [
    USE WALL_SHAPE
  ]
  boundingObject USE WALL_SHAPE
}
DEF Cozmo Robot {
  translation 0 0.016 0
  rotation 9.30088e-10 -1.52937e-07 -1 7.48661e-07
  children [
    DEF BODY Shape {
      appearance Appearance {
        material DEF MAT_WHITE Material {
          diffuseColor 1 1 1
        }
      }
      geometry Box {
        size 0.014 0.014 0.048
      }
    }
    DEF WHEEL_REAR_RIGHT HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.02 0 0.024
      }
      endPoint Solid {
        translation 0.02 3.92458e-06 0.024
        rotation 1 0 0 0
        children [
          DEF WHEEL Transform {
            rotation 0 0 1 1.5708
            children [
              Shape {
                appearance Appearance {
                  material Material {
                    diffuseColor 0.254032 0.254032 0.254032
                  }
                }
                geometry Cylinder {
                  height 0.02
                  radius 0.015
                  subdivision 36
                }
              }
            ]
          }
        ]
        contactMaterial "nofriction"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL_REAR_LEFT HingeJoint {
      jointParameters HingeJointParameters {
        position -1.15762e-11
        anchor -0.02 0 0.024
      }
      endPoint Solid {
        translation -0.02 3.92342e-06 0.024
        rotation 0.0797287 -2.06116e-06 0.996817 -1.15774e-11
        children [
          USE WHEEL
        ]
        contactMaterial "nofriction"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL_FRONT_RIGHT HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.02 0 -0.024
      }
      device RotationalMotor {
        name "wheel_fr"
      }
      endPoint Solid {
        translation 0.02 3.92458e-06 -0.024
        rotation 1 0 0 0
        children [
          USE WHEEL
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL_FRONT_LEFT HingeJoint {
      jointParameters HingeJointParameters {
        position -1.15794e-11
        anchor -0.02 0 -0.024
      }
      device RotationalMotor {
        name "wheel_fl"
      }
      endPoint Solid {
        translation -0.02 3.92342e-06 -0.024
        rotation 0.0794063 -2.05874e-06 0.996842 -1.15799e-11
        children [
          USE WHEEL
        ]
        boundingObject USE WHEEL
        physics Physics {
          centerOfMass [
            0 0 0
          ]
        }
      }
    }
    DEF HEAD HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0 0.02 0
      }
      device RotationalMotor {
        name "motor_head_pitch"
      }
      endPoint Solid {
        rotation 1 0 0 0
        children [
          DEF CAM_HEAD Camera {
            translation 0 0.028 -0.025
            rotation 1 0 0 0
            children [
              Transform {
                rotation 1 0 0 -0.7
                children [
                  SpotLight {
                    beamWidth 0.25
                    cutOffAngle 0.8
                    direction 0 1 -1
                    intensity 0.8
                  }
                ]
              }
            ]
            name "cam_head"
            width 640
            height 480
            near 0.001
            pixelSize 0.5
          }
          DEF EYE_LED LED {
            translation 0.013 0.034 -0.03
            rotation 1 0 0 1.5708
            children [
              Shape {
                appearance Appearance {
                  material Material {
                    diffuseColor 0.392905 0.611307 0.8
                  }
                }
                geometry Cylinder {
                  height 0.001
                  radius 0.009
                  subdivision 24
                }
              }
            ]
            name "LED_LEFT_EYE"
            color [
              0 0.510063 1
            ]
          }
          DEF EYE_LEFT Transform {
            translation -0.013 0.034 -0.01
            rotation 1 0 0 -1.5708
            children [
              DEF EYE_SHAPE Shape {
                appearance Appearance {
                  material Material {
                    diffuseColor 1 1 1
                  }
                }
                geometry Cylinder {
                  height 0.04
                  radius 0.012
                  subdivision 24
                }
              }
            ]
          }
          DEF EYE_RIGHT Transform {
            translation 0.013 0.034 -0.01
            rotation 1 0 0 -1.5708
            children [
              USE EYE_SHAPE
            ]
          }
        ]
      }
    }
  ]
  name "Cozmo"
  boundingObject USE BODY
  physics Physics {
    density -1
    mass 1
  }
  controller "cozmo_basicdrive01"
}
DEF Block1x1 Solid {
  translation 0.1 0.031 -0.08
  rotation 0.00718211 -0.999848 0.0158744 1.28749e-15
  children [
    Transform {
      children [
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "/Users/hanns/Documents/Cozmo01/code01.png"
              ]
            }
          }
          geometry Box {
            size 0.06001 0.038 0.038
          }
        }
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "/Users/hanns/Documents/Cozmo01/code01.png"
              ]
            }
          }
          geometry Box {
            size 0.038 0.06001 0.038
          }
        }
        Shape {
          appearance Appearance {
            texture ImageTexture {
              url [
                "/Users/hanns/Documents/Cozmo01/code01.png"
              ]
            }
          }
          geometry Box {
            size 0.038 0.038 0.06001
          }
        }
      ]
    }
    DEF Block1x1Shape Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 0 0
        }
      }
      geometry Box {
        size 0.06 0.06 0.06
      }
    }
  ]
  boundingObject USE Block1x1Shape
  physics DEF Block1x1Physics Physics {
  }
}
Solid {
  translation -0.2 0.2 0
  children [
    Shape {
      geometry DEF BottomRamp Box {
        size 0.06 0.001 0.09
      }
    }
  ]
  boundingObject USE BottomRamp
  physics Physics {
  }
}
